start-from-zero/Javascript/src/data/storeData.js

```diff
@@ -1,45 +1,50 @@
-windowObjectWrap.FormData = function(){
-    btn.addEventListener('click',function(){
-        function User(username,date,radio,textarea,select){
-        this.username = username;
-        this.date = date;
-        this.radio = radio;
-        this.textarea = textarea.forEach(element=>{
-            this[element.name] = element.value;
-        });
-        this.select = select.forEach(element=>{
-            this[element.name] = element.value;
-        })
-        }
-
-        const username = document.getElementById('username-id').value;
-        const calendar = document.getElementById('calendar-id').value;
-        const checkbox = document.querySelectorAll('.input-checkbox');
-        const textarea = document.querySelectorAll('textarea');
-        const legend = document.querySelectorAll('legend');
-        const select = document.querySelectorAll('select');
-    
-        const radioArr = [];
-        checkbox.forEach(element => {
-            if(element.checked === true){
-                radioArr.push(element.value);
+(function(){
+    windowObjectWrap.FormData = function(){
+        btn.addEventListener('click',function(){
+            function User(username,date,radio,textarea,select,id){
+            this.username = username;
+            this.date = date;
+            this.radio = radio;
+            this.textarea = textarea.forEach(element=>{
+                this[element.name] = element.value;
+            });
+            this.select = select.forEach(element=>{
+                this[element.name] = element.value;
+            });
+            this.id = (this.username + new Date().getTime()).replace(/ /g, '').toLowerCase();
```

Comment:
This one's a nice one as well. If you can explain it, that is! :D

Reply:
I created an unique id for all users. I took the username value and concatenated with the time that was created(the number of miliseconds that have passed since 1970), because I wanted to assure that the id will be unique. I wrapped the concatenated user and date and removed the empty spaces with replace(/ /g, ''). After that I transformed everything into lower case because I wanted to use this id as custom data attribute. In the mdn documentation it says that the custom data attribute must not contain ASCII capital letters (A to Z).